### PR TITLE
feat(dev-server): introduce `env`

### PR DESCRIPTION
**Pull request by yusukebe**
This PR introduces the concept of `ENV` and the `env` option.

For Cloudflare Pages, the current `cf` option allows us to use bindings. For example:

```ts
export default defineConfig({
  plugins: [
    devServer({
      entry: 'src/index.ts',
      cf: {
        bindings: {
          NAME: 'Hono',
        },
      },
    }),
  ],
})
```

With this configuration, we can access variables through `c.env`:

```ts
app.get('/', (c) => {
  return c.text(c.env.NAME)
})
```

This is currently exclusive to Cloudflare Pages, but in the future, it may be extended to other runtimes/platforms with specific environment values. While `c.env` seems to be used for Cloudflare, other platforms/runtimes may also use bindings/`c.env`. For instance, the AWS Lambda adapter currently uses `env`:

https://github.com/honojs/hono/blob/main/src/adapter/aws-lambda/handler.ts#L126-L143

This PR suggests separating the Cloudflare Pages-specific functionality from `dev-server.ts`. With this change, the `getEnv()` method for Cloudflare Pages is placed in `cloudflare-pages/index.ts`, and users can include the `env` in their `vite.config.ts`:

```ts
import { getEnv } from '@hono/vite-dev-server/cloudflare-pages'

// ...

export default defineConfig({
  plugins: [
    devServer({
      env: getEnv({
        bindings: {
          NAME: 'Hono',
        },
      }),
    }),
  ],
})
```

With this mechanism, adding support for another platform that uses environment variables becomes straightforward:

```ts
import { getEnv } from '@hono/vite-dev-server/another-one'

// ...

export default defineConfig({
  plugins: [
    devServer({
      env: getEnv({
        someFunction: {
          NAME: 'Hono',
        },
      }),
    }),
  ],
})
```

### CODE REVIEW

**Comment by yusukebe** [comment]
Hey @Code-Hex !

Could you review this?

File: packages/dev-server/src/dev-server.ts [hunk 1]
```diff
@@ -1,29 +1,20 @@
...
-        MiniflareOptions,
-        'cachePersist' | 'd1Persist' | 'durableObjectsPersist' | 'kvPersist' | 'r2Persist'
-      > & {
-        // Enable `env.ASSETS.fetch()` function for Cloudflare Pages.
-        assets?: boolean
-      }
-  >
+  env?: Env | EnvFunc
+} & {
+  // The `cf` option is for backward compatibility. Typically, you should use `env`.
```
Comment (Code-Hex) [inline packages/dev-server/src/dev-server.ts:13]: Is this better to add `@deprecate` JSDoc?

**Review by Code-Hex — APPROVED** [review]
Looks great to me 😁

File: packages/dev-server/src/dev-server.ts (hunk 1 above)
Comment (yusukebe) [reply packages/dev-server/src/dev-server.ts]: Yeah. I've added.

**Comment by yusukebe** [comment]
@Code-Hex 

Thanks for reviewing. Let's go with this!

**Pull request merged**